pacai/util/distanceCalculator.py

```diff
@@ -115,7 +115,7 @@ def computeDistances(layout):
         for node in allNodes:
             dist[node] = sys.maxsize
 
-        queue = util.PriorityQueue()
+        queue = priority_queue.PriorityQueue()
```

Comment:
I believe the distance calculator class is essentially a cache and the distancer class is set up to work as an interface for the cache. Initially, the cache is filled with the manhattan distances between the positions on the map layout that Pacman or a Ghost can be on. When distancer.getMazeDistances() is called, then UCS is performed to compute the maze distances on the map layout using the computeDistances() function. I believe they could just be one class.

Secondly, getDistance() is used only for CaptureAgents, it's needed when a Capture Agent calls getMazeDistance(). The UCS function is called by the CaptureAgents in the registerInitialState function, when self.distancer.getMazeDistances() is called.

Reply:
Perhaps a solution to hiding the UCS code is to have computeDistances() create a PositionSearchProblem and call the UCS function that the students have to implement. The primary reason I suggest this is because if I recall correctly that CaptureAgents are only used in the contest. Which is far later in the quarter. I see one problem with this which is that the initial game state for the PositionSearchProblem is the game state defined in pacman.py